### PR TITLE
Feat/sequence scrollbar

### LINKED ISSUE #792: [Feature] support scrollbar in sequence chart
### What problem does this feature solve?

when I scroll mouse, the canvas or axes will be scaled, I don't need scale, I just want to scroll canvas. 
Maybe set the scrollbar to visible can solve this problem, but it not support scrollbar in sequence chart.

### What does the proposed API look like?

null

### PR DESCRIPTION
**Pull request by skie1997**
[[中文版模板 / Chinese template](https://github.com/VisActor/VChart/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #792 

### 🐞 Bugserver case id
64f74ac12937835bd5c893ec

### 💡 Background and solution
20230906225524_rec_

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/chart/sequence/sequence.ts [hunk 1]
```diff
@@ -178,6 +182,29 @@ export class SequenceChart extends BaseChart {
...
 
+        // scrollBar 内置
+        if (seriesSpec.type === SeriesTypeEnum.dot) {
+          elements.push({
+            modelId: `scrollBarRightRow${rowNum}`,
+            col: 2,
+            row: rowNum
+          });
+          const data = this.getSeriesData(seriesSpec.dataId, seriesSpec.dataIndex);
+          const ratio = (seriesSpec?.height || defaultSeriesRowHeight) / (data.latestData.length * 30);
```
Comment (kkxxkk2019) [inline packages/vchart/src/chart/sequence/sequence.ts:193]: 数据有没有可能为空？

File: packages/vchart/src/chart/sequence/sequence.ts [hunk 2]
```diff
@@ -12,8 +12,11 @@ import type { Datum } from '../../typings';
 import type { DataView } from '@visactor/vdataset';
 import type { ISeriesOption } from '../../series/interface';
 import { VChart } from '../../core/vchart';
+import type { ICartesianAxisSpec, IScrollBarSpec } from '../../component';
 VChart.useSeries([DotSeries, LinkSeries]);
 
+const SCROLL_BAR_DEFAULT_SIZE = 10;
```
Comment (kkxxkk2019) [inline packages/vchart/src/chart/sequence/sequence.ts:18]: 这个从主题拿会不会合适些？

File: packages/vchart/src/chart/sequence/sequence.ts [hunk 3]
```diff
@@ -50,7 +54,7 @@ export class SequenceChart extends BaseChart {
       (spec as any).legends[0].id = `legendRow${rowNum}`;
       rowHeight.push({
         index: rowNum,
-        size: 20
+        size: 40
```
Comment (kkxxkk2019) [inline packages/vchart/src/chart/sequence/sequence.ts:57]: 这个是写死的吗？用户可不可以配置？

File: packages/vchart/src/chart/sequence/sequence.ts (hunk 1 above)
Comment (skie1997) [reply packages/vchart/src/chart/sequence/sequence.ts:193]: 有可能，这里我做下处理

File: packages/vchart/src/chart/sequence/sequence.ts (hunk 2 above)
Comment (skie1997) [reply packages/vchart/src/chart/sequence/sequence.ts:18]: 嗯嗯，这个size应该可以作为通用的主题配置

File: packages/vchart/src/chart/sequence/sequence.ts (hunk 3 above)
Comment (skie1997) [reply packages/vchart/src/chart/sequence/sequence.ts:57]: 我想想，这里可以取legend的height作为配置的，我调整下

File: packages/vchart/src/chart/sequence/sequence.ts (hunk 2 above)
Comment (skie1997) [reply packages/vchart/src/chart/sequence/sequence.ts]: 补充：scrollbar是用width和height作为size的配置的，跟orient有关，写死在主题里面的话不能应对不同orient的场景，所以还是不能放在主题里面

File: packages/vchart/src/chart/sequence/sequence.ts [hunk 4]
```diff
@@ -41,27 +44,27 @@ export class SequenceChart extends BaseChart {
     const leftAppendPadding = spec?.appendPadding?.left || 0;
     const rightAppendPadding = spec?.appendPadding?.right || 0;
 
-    if ((spec as any)?.legends) {
+    if (spec?.legends) {
```
Comment (kkxxkk2019) [inline packages/vchart/src/chart/sequence/sequence.ts:47]: 这里需要处理下，legends 有可能是对象

**Review by kkxxkk2019 — APPROVED** [review]

File: packages/vchart/src/chart/sequence/sequence.ts (hunk 4 above)
Comment (skie1997) [reply packages/vchart/src/chart/sequence/sequence.ts:47]: ok

**Review by kkxxkk2019 — APPROVED** [review]

**Pull request merged**